app/assets/javascripts/backbone/views/form_items_view.js.coffee

```diff
@@ -0,0 +1,93 @@
+class ELMO.Views.FormItemsView extends Backbone.View
+
+  el: '.form-items'
+
+  events:
+    'click .add-group': 'show_new_group_modal'
+    'click .form-group .edit': 'show_edit_group_modal'
+    'click .form-group .delete': 'delete_group'
+
+  initialize: (params) ->
+    this.nested_list()
+    this.params = params
+
+  show_new_group_modal: (link) ->
+    link.preventDefault()
+    ELMO.app.loading(true)
+
+    $.ajax({
+      url: ELMO.app.url_builder.build('qing-groups', 'new'),
+      method: "get",
+      data: {form_id: this.params.form_id},
+      success: (html) =>
+        new ELMO.Views.GroupModalView({html: html, list_view: this, mode: 'new'})
+        ELMO.app.loading(false)
+    })
+
+  show_edit_group_modal: (link) ->
+    link.preventDefault()
+    $link = $(link.currentTarget)
```

Comment:
Clever!

On 13 March 2015 at 10:18, Elizabeth Mitchell <notifications@github.com>
wrote:

> In app/assets/javascripts/backbone/views/form_items_view.js.coffee
> <https://github.com/thecartercenter/elmo/pull/44#discussion_r26388499>:
>
> > +  show_new_group_modal: (link) ->
> > +    link.preventDefault()
> > +    ELMO.app.loading(true)
> > +
> > +    $.ajax({
> > +      url: ELMO.app.url_builder.build('qing-groups', 'new'),
> > +      method: "get",
> > +      data: {form_id: this.params.form_id},
> > +      success: (html) =>
> > +        new ELMO.Views.GroupModalView({html: html, list_view: this, mode: 'new'})
> > +        ELMO.app.loading(false)
> > +    })
> > +
> > +  show_edit_group_modal: (link) ->
> > +    link.preventDefault()
> > +    $link = $(link.currentTarget)
>
> $item means $(item), so that I don't use $() more times than needed. It
> means that it is a jQuery object and jQuery doesn't have to be called.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/thecartercenter/elmo/pull/44/files#r26388499>.
>



-- 
Tom Smyth

Worker-Owner, Sassafras Tech Collective
Specializing in innovative, usable tech for social change
sassafras.coop *·* @sassafrastech

Resident, Touchstone Cohousing
touchstonecohousing.org